_posts/2020-04-08-secure-software-development-lifecycle.md

```diff
@@ -0,0 +1,138 @@
+
+#### Requirements
+
+There are laws and regulations. There are company policies and customer policies. Failing to notice these will make it difficult to deal with them later. Gathering intelligence on what you are about to do businesswise should make you able to also make some security requirements for the project. This will create the foundation of your security awareness level later on. Are you dealing with credit cards? Surely you have then heard about PCI-DSS. Are you creating an application for healthcare? Maybe you ought to know something about HIPAA. Know your industry and make use of whatever given requirements there are for the field. You should also have a hunch about data classification levels.
+
+To sum up: 
+* Know the laws and regulations
+* Follow company policies for your own company and for your customer company
+* Check for useful standards
+* Make security requirements
+* Data classification
+
+#### Design
+
+Design sets a foundation for software architecture. Security is all about CIA (confidentiality, integrity and availability). Failures in software architecture most often affects one of the letters of CIA and they are also often pretty expensive to fix. To fight this we would like to setup extra care for design. There are plenty of tools and practices in different software project management frameworks but I would like to bring out threat modeling. There are different approaches for threat modeling but what I have found useful is to create diagrams for software architecture from different approaches: deployment pipeline, infrastructure and networking, sequence diagrams and workflows. Reviewing those diagrams from the "how would I hack this?" -perspective is really useful. You can also go so far that you create evil user stories for the backlog. 
+
+To sum up:
+* Create architecture diagrams and review them
+* Threat modeling 
+* Evil user stories
+
+
+#### Implementation
+
+Implementation is where the stuff that developers love happens. There are numerous ways to improve security or fail bad at this phase. Most important thing will be your developers security awareness. Many of the security issues are more or less stuff that nobody came to think about. Yes, I mean the OWASP Top 10 stuff like SQL injections. One thing you need to understand about developers is that they love to solve problems. So normal trick is to try to dress security requirements as a solvable problems like "make sure that all input is sanitized against SQL injections" and "make sure that all output is sanitized against XSS". This is important but of course covers only one problem at a time and it will become a pretty exhaustive list to implement all over again in every project. Oftentimes more eyes at the same problem will help in finding loopholes in your security design. 
```

Comment:
Most important thing will be <- "The most important thing" or rephrase to raising your developers' security awareness?